srcts/src/time/throttle.ts

```diff
@@ -46,13 +44,28 @@ class Throttler<X extends AnyVoidFunction> implements InputRatePolicy<X> {
       this.timerId = null;
     }
   }
+  // Invoke the throttled function with the currently-stored args
+  // and set the timeout
   $invoke(): void {
-    if (this.args && this.args.length > 0) {
-      this.func.apply(this.target, this.args);
-    } else {
-      this.func.apply(this.target);
-    }
-    this.args = null;
+    // Empty list on a function that doesn't expect arguments will work fine
+    this.func.apply(this.target, this.args);
+
+    this.args = null; // clear the stored args
+
+    // set this.timerId to a newly-created timer, which will invoke a call
+    // with the most recently called args (if any) when it expires
+    this.timerId = setTimeout(() => {
+      // IE8 doesn't reliably clear timeout, so this additional
+      // check is needed
+      if (this.timerId === null) return;
+
+      this.$clearTimer();
+      // Do we have a call queued up? (check existence rather than length)
+      if (this.args) {
```

Comment:
Can you make this explicit? That will help reduce the possibility of confusion in the future about what counts as falsy.

```suggestion
      if (this.args === null) {
```

Reply:
Actually, now that I think about it, I still think this code could lead to confusion.

Specifically, when `args` is an array, that means that a call is scheduled, and when `args` is `null` means that no call is scheduled.

Elsewhere, in `normalCall`, this uses `this.timerId === null` to check whether a call is scheduled.

It would be clearer to consistently use the same condition.

Reply:
Actually, those are intentionally testing different things. `timerId` checks whether there was an invocation within the last interval; `args` checks if there’s a call waiting.

Reply:
Making the bill check explicit seems wise though

Reply:
Oh `isPending()` should be using `this.args` I'll fix that